server/stream.go

```diff
@@ -4174,8 +4200,8 @@ func (mset *stream) getDirectRequest(req *JSApiMsgGetRequest, reply string) {
 
 	// If batch was requested send EOB.
 	if batch > 1 {
-		np, _ := store.NumPending(seq, req.NextFor, false)
-		hdr := []byte(fmt.Sprintf("NATS/1.0 204 EOB\r\nNats-Pending-Messages: %d\r\n\r\n", np))
+		// TODO(dlc) - Should we recalculate num pending here to account for new messages?
```

Comment:
yeah would feel weird if these are wrong

Reply:
Chance os low but not zero. I could re-calculate on EOB marker but again those calls aren't free. Or I could add ValidThrough like we do with some lower level calls.